File: sample-java/src/main/java/social/bigbone/sample/GetUserProfile.java
package social.bigbone.sample;

import social.bigbone.MastodonClient;
import social.bigbone.api.entity.Profile;
import social.bigbone.api.exception.BigBoneRequestException;

public class GetUserProfile {
    public static void main(final String[] args) throws BigBoneRequestException {
        final String instance = args[0];

        // access token with at least Scope.READ.ACCOUNTS or Scope.PROFILE.ALL
        final String accessToken = args[1];

        // Instantiate client
        final MastodonClient client = new MastodonClient.Builder(instance).accessToken(accessToken).build();

        // Get and display user profile
        final Profile profile = client.profile().getUserProfile().execute();

        System.out.println(profile.getDisplayName());
        System.out.println(profile.getId());
        System.out.println(profile.getNote());
        for (Profile.Field field : profile.getFields()) {
            System.out.print(field.getName());
            System.out.print(": ");
            System.out.println(field.getValue());
        }
    }
}

File: sample-kotlin/src/main/kotlin/social/bigbone/sample/GetUserProfile.kt
package social.bigbone.sample

import social.bigbone.MastodonClient

object GetUserProfile {
    @JvmStatic
    fun main(args: Array<String>) {
        val instance = args[0]

        // access token with at least Scope.READ.ACCOUNTS or Scope.PROFILE.ALL
        val accessToken = args[1]

        // instantiate client
        val client = MastodonClient.Builder(instance)
            .accessToken(accessToken)
            .build()

        // Get and display user profile
        val profile = client.profile.getUserProfile().execute()
        profile.apply {
            print("$displayName (ID: $id)\n")
            print("--------------------\n")
            print("$note\n")
            print("--------------------\n")
            for (field in fields) {
                print("${field.name}: ${field.value}\n")
            }
        }
    }
}
